Sources/mpc-core-kit-swift/MpcCoreKitSwift.swift

```diff
@@ -15,129 +15,113 @@ import TorusUtils
 public class MpcCoreKit {
     internal var selectedTag: String?
     internal var factorKey: String?
+    internal var tssShare: String?
+    internal var tssIndex: String?
 
-    // TODO: Replace [String: Any] with [String: Codable], throughout
-    internal var userInfo: [String: Any]?
+    internal var tssPubKey: String?
 
-    internal var oauthKey: String?
-    internal var network: Web3AuthNetwork
-    internal var option: CoreKitOptions
+    internal var userInfo: UserInfo?
+    internal var option: CoreKitWeb3AuthOptions
+    // internal var state: CoreKitstate
 
-    internal var appState: CoreKitAppState
+    public var metadataPubKey: String?
+    public var oAuthKey: String?
+    // share index used for backup share recovery
+    public var deviceMetadataShareIndex: String?
+    public var loginTime: Date?
 
     public var metadataHostUrl: String?
-
     public var tkey: ThresholdKey?
-
     public var tssEndpoints: [String]?
     public var authSigs: [String]?
     public var verifier: String?
     public var verifierId: String?
-
     public var torusUtils: TorusUtils
     public var nodeIndexes: [Int]?
     public var nodeDetails: AllNodeDetailsModel?
-
     public var nodeDetailsManager: NodeDetailManager
-
     public var sigs: [String]?
 
     public var coreKitStorage: CoreKitStorage
-
     private let storeKey = "corekitStore"
+    private let localstateKey = "localstate"
+    private let customAuth: CustomAuth
 
-    private let localAppStateKey = "localAppState"
-
-    // init
-    // TODO: This should accept a param class instead
-    public init(web3AuthClientId: String, web3AuthNetwork: Web3AuthNetwork, disableHashFactor: Bool = false, localStorage: ILocalStorage, manualSync: Bool = false) throws {
-        if web3AuthClientId.isEmpty {
+    public init(options: CoreKitWeb3AuthOptions) throws {
```

Comment:
we should add default options similar to web in case of optional params, for ex if network is not provided thn it should fallback to sapphire_mainnet etc

Reply:
@ieow